src/codecs/avif/encoder.ts

```diff
@@ -0,0 +1,18 @@
+import avif_enc, { AVIFModule } from '../../../codecs/avif_enc/avif_enc';
+import wasmUrl from '../../../codecs/avif_enc/avif_enc.wasm';
+import { EncodeOptions } from './encoder-meta';
+import { initEmscriptenModule } from '../util';
+
+let emscriptenModule: Promise<AVIFModule>;
+
+export async function encode(data: ImageData, options: EncodeOptions): Promise<ArrayBuffer> {
+  if (!emscriptenModule) emscriptenModule = initEmscriptenModule(avif_enc, wasmUrl);
+
+  const module = await emscriptenModule;
+  const resultView = module.encode(data.data, data.width, data.height, options);
+  const result = new Uint8Array(resultView);
+  module.free_result();
+
+  // wasm can’t run on SharedArrayBuffers, so we hard-cast to ArrayBuffer.
+  return result.buffer as ArrayBuffer;
```

Comment:
Same (although now I've looked through other codecs and I see that we do this everywhere 😶).

Reply:
Actually no, this one is fine because we create a new copy of `Uint8Array` unlike in `ImageData` cases.